pkg/config/config.go

```diff
@@ -247,8 +247,18 @@ func setModelServerReqEndpoint() string {
 
 // return local path to power model weight
 // e.g., /var/lib/kepler/data/model_weight/acpi_AbsPowerModel.json
-func GetDefaultPowerModelURL(modelOutputType, energySource string) string {
-	return fmt.Sprintf(`/var/lib/kepler/data/model_weight/%s_%sModel.json`, energySource, modelOutputType)
+func GetDefaultPowerModelURL(modelOutputType, energySource, cpuArch string) string {
+	// strip white space or new line from cpuArch
+	cpuArch = strings.TrimSuffix(cpuArch, "\n")
+	cpuArch = strings.TrimSpace(cpuArch)
+	fullPath := fmt.Sprintf(`/var/lib/kepler/data/%s/model_weight/%s_%sModel.json`, cpuArch, energySource, modelOutputType)
+	// if the model does not exist, return the default model
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
```

Comment:
On a minor note, os.isNotExist(err) might be misleading as os.Stat can return other types of errors which is not the same as a file/model not existing.